spec/models/provider_spec.rb

```diff
@@ -214,70 +216,86 @@
     end
   end
 
-  context 'AGFS supplier number validation for a chamber' do
-    context 'for a blank supplier number' do
+  context 'when validating AGFS supplier number for a chamber' do
+    context 'with a blank supplier number' do
+      before { chamber.firm_agfs_supplier_number = '' }
+
       it 'returns no errors' do
-        chamber.firm_agfs_supplier_number = ''
         expect(chamber).to be_valid
       end
```

Comment:
(nit)

```ruby
it { is_expected.to be_valid }
```

or

```ruby
it 'returns no errors' do
  expect { chamber.save }.not_to raise_error
end
```

I'd favour the first as the current version isn't actually testing "returns no errors".

Reply:
There is a lot more i could have done but need to stop yak shaving.